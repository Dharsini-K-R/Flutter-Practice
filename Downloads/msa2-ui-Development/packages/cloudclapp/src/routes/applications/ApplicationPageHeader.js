import React from "react";
import { Grid, Typography } from "@material-ui/core";
import { CubeIcon } from "react-line-awesome";
import { useTranslation } from "react-i18next";
import { useCommonStyles } from "cloudclapp/src/styles/commonStyles";

const ApplicationPageHeader = () => {
  const commonClasses = useCommonStyles();
  const { t } = useTranslation();
  return (
    <div
      className={commonClasses.commonPageHeaderContainer}
      data-testid="applications-container"
    >
      <Grid
        container
        className={commonClasses.commonPageHeaderGrid}
        spacing={2}
      >
        <Grid item>
          <CubeIcon className={commonClasses.commonPageHeaderIcon} />
        </Grid>
        <Grid item>
          <Typography
            id="applications_title"
            variant="h4"
            className={commonClasses.commonPageHeaderText}
          >
            {" "}
            {t("Application Images")}
          </Typography>
        </Grid>
      </Grid>
    </div>
  );
};

export default ApplicationPageHeader;
