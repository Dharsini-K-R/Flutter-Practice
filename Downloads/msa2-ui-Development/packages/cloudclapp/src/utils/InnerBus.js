import EventEmitter from "events";

// List of all events available for usage with inner bus
const EVENTS = {
  LOGOUT: "LOGOUT",
  LOGIN: "LOGIN",
};

class InnerBus extends EventEmitter {
  evt = EVENTS;
}

export default new InnerBus();
