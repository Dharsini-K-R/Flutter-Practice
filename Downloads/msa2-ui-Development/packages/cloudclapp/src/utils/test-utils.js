import React from "react";
import { render } from "@testing-library/react";
import { renderHook, act } from "@testing-library/react-hooks";
import userEvent from "@testing-library/user-event";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import { BrowserRouter as Router } from "react-router-dom";
import { StylesProvider } from "@material-ui/core";
import { ThemeProvider } from "@material-ui/core";
import SnackbarProvider from "cloudclapp/src/SnackbarProvider";
import createTheme from "cloudclapp/src/styles/theme";
import thunk from "redux-thunk";
const middlewares = [thunk];

const msaTheme = createTheme();

// Make generated classNames consistent for snapshots
const generateClassName = (rule, styleSheet) =>
  `${styleSheet.options.classNamePrefix}-${rule.key}`;

const AllTheProviders = ({ children }) => {
  return (
    <StylesProvider generateClassName={generateClassName}>
      <ThemeProvider theme={msaTheme}>
        <SnackbarProvider>{children}</SnackbarProvider>
      </ThemeProvider>
    </StylesProvider>
  );
};

const customRender = (component, options, initialRoute = "/") => {
  const componentWithRouter = (
    <Router basename={initialRoute}>{component}</Router>
  );

  return render(componentWithRouter, { wrapper: AllTheProviders, ...options });
};

const renderWithProvider = ({
  childComponent,
  mockStore,
  mockState = {},
  initialRoute,
}) => {
  const store = mockStore || configureMockStore(middlewares)(mockState);

  return {
    ...customRender(
      <Provider store={store}>{childComponent}</Provider>,
      null,
      initialRoute,
    ),
    store,
  };
};

const customRenderHook = (callback, options = {}) => {
  const wrapper = options.wrapper || AllTheProviders;

  return renderHook(callback, { ...options, wrapper });
};

// re-export everything
export * from "@testing-library/react";

// override render and renderHook methods
export {
  customRender as render,
  renderWithProvider,
  customRenderHook as renderHook,
  act as actOnHook,
  userEvent,
};
