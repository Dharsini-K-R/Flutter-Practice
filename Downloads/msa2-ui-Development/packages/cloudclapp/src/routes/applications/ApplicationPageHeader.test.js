import React from "react";
import { render } from "cloudclapp/src/utils/test-utils";
import ApplicationPageHeader from "./ApplicationPageHeader";

describe("Application Page Header component", () => {
  it("renders correctly", () => {
    const { container, getByText, getByTestId } = render(
      <ApplicationPageHeader />,
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(getByTestId("applications-container").toBeTruthy);
    expect(getByText("Application Images")).toBeTruthy();
  });
});
