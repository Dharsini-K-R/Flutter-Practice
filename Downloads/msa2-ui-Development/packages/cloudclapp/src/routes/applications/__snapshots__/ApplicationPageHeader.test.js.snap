// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application Page Header component renders correctly 1`] = `
<div
  class="makeStyles-commonPageHeaderContainer"
  data-testid="applications-container"
>
  <div
    class="MuiGrid-root makeStyles-commonPageHeaderGrid MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <i
        aria-hidden="true"
        class="la la-cube makeStyles-commonPageHeaderIcon"
        role="presentation"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <h4
        class="MuiTypography-root makeStyles-commonPageHeaderText MuiTypography-h4"
        id="applications_title"
      >
         
        Application Images
      </h4>
    </div>
  </div>
</div>
`;
